board/safety/safety_volkswagen.h

```diff
@@ -1,142 +1,158 @@
-AddrCheckStruct volkswagen_rx_checks[] = {
+AddrCheckStruct volkswagen_mqb_rx_checks[] = {
   {.addr = {MSG_EPS_01}, .bus = 0, .expected_timestep = 10000U},
   {.addr = {MSG_ACC_06}, .bus = 0, .expected_timestep = 20000U},
   {.addr = {MSG_MOTOR_20}, .bus = 0, .expected_timestep = 20000U},
 };
+const int VOLKSWAGEN_MQB_RX_CHECKS_LEN = sizeof(volkswagen_mqb_rx_checks) / sizeof(volkswagen_mqb_rx_checks[0]);
 
-const int VOLKSWAGEN_RX_CHECK_LEN = sizeof(volkswagen_rx_checks) / sizeof(volkswagen_rx_checks[0]);
 
-struct sample_t volkswagen_torque_driver;  // last few driver torques measured
+struct sample_t volkswagen_torque_driver; // Last few driver torques measured
 int volkswagen_rt_torque_last = 0;
 int volkswagen_desired_torque_last = 0;
 uint32_t volkswagen_ts_last = 0;
 int volkswagen_gas_prev = 0;
+int volkswagen_torque_msg = 0;
+int volkswagen_lane_msg = 0;
+
+static void volkswagen_mqb_init(int16_t param) {
+  UNUSED(param);
+
+  controls_allowed = false;
+  relay_malfunction = false;
+  volkswagen_torque_msg = MSG_HCA_01;
+  volkswagen_lane_msg = MSG_LDW_02;
+
+  return;
+}
 
-static int volkswagen_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
+static int volkswagen_mqb_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
 
-  bool valid = addr_safety_check(to_push, volkswagen_rx_checks, VOLKSWAGEN_RX_CHECK_LEN,
+  bool valid = addr_safety_check(to_push, volkswagen_mqb_rx_checks, VOLKSWAGEN_MQB_RX_CHECKS_LEN,
                                  NULL, NULL, NULL);
 
   if (valid) {
-   int bus = GET_BUS(to_push);
-   int addr = GET_ADDR(to_push);
-
-   // Update driver input torque samples from EPS_01.Driver_Strain for absolute torque, and EPS_01.Driver_Strain_VZ
-   // for the direction.
-   if ((bus == 0) && (addr == MSG_EPS_01)) {
-     int torque_driver_new = GET_BYTE(to_push, 5) | ((GET_BYTE(to_push, 6) & 0x1F) << 8);
-     int sign = (GET_BYTE(to_push, 6) & 0x80) >> 7;
-     if (sign == 1) {
-       torque_driver_new *= -1;
-     }
-
-     update_sample(&volkswagen_torque_driver, torque_driver_new);
-   }
-
-   // Monitor ACC_06.ACC_Status_ACC for stock ACC status. Because the current MQB port is lateral-only, OP's control
-   // allowed state is directly driven by stock ACC engagement. Permit the ACC message to come from either bus, in
-   // order to accommodate future camera-side integrations if needed.
-   if (addr == MSG_ACC_06) {
-     int acc_status = (GET_BYTE(to_push, 7) & 0x70) >> 4;
-     controls_allowed = ((acc_status == 3) || (acc_status == 4) || (acc_status == 5)) ? 1 : 0;
-   }
-
-   // exit controls on rising edge of gas press. Bits [12-20)
-   if (addr == MSG_MOTOR_20) {
-     int gas = (GET_BYTES_04(to_push) >> 12) & 0xFF;
-     if ((gas > 0) && (volkswagen_gas_prev == 0)) {
-       controls_allowed = 0;
-     }
-     volkswagen_gas_prev = gas;
-   }
-
-   if ((safety_mode_cnt > RELAY_TRNS_TIMEOUT) && (bus == 0) && (addr == MSG_HCA_01)) {
-     relay_malfunction = true;
-   }
+    int bus = GET_BUS(to_push);
```

Comment:
GitHub thinks there are whitespace diffs here. are there?

Reply:
ah yes, now is correct :)

Reply:
Yeah, there was a whole block indented by one space instead of two in the prior code.